core/server/master/src/main/java/alluxio/master/job/UfsFileIterable.java

```diff
@@ -121,13 +119,13 @@ private void partialListFileInfos() {
     private void listFileInfos() {
       try {
         AuthenticatedClientUser.set(mUser.orElse(null));
-        UfsStatus[] ufsStatuses = mFs.listStatus(mPath);
-        if (ufsStatuses == null || ufsStatuses.length == 0) {
+        Optional<UfsStatus[]> ufsStatuses = mFs.listStatuses(mPath, ListOptions.defaults());
+        if (!ufsStatuses.isPresent() || ufsStatuses.get().length == 0) {
```

Comment:
It's OK to write the conditions like this in C/C++ programming, because 
If `ufsStatuses.isPresent()` is false, the `ufsStatuses.get().length == 0` will not be evaluated.

I bet Java programming language does the same, right?

Reply:
emmmm... java may not lol
it needs to validate existing first, otherwise NPE may throw